jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp

```diff
@@ -57,21 +57,29 @@
  */
 static ecma_completion_value_t
 ecma_builtin_array_prototype_helper_set_length (ecma_object_t *object, /**< object*/
-                                                uint32_t length) /**< new length */
+                                                ecma_number_t length) /**< new length */
 {
   ecma_completion_value_t ret_value;
-  ecma_string_t *magic_string_length_p = ecma_get_magic_string (LIT_MAGIC_STRING_LENGTH);
 
-  ecma_number_t *len_p = ecma_alloc_number ();
-  *len_p = ecma_uint32_to_number (length);
+  if (ecma_object_get_class_name (object) == LIT_MAGIC_STRING_ARRAY_UL
+      && unlikely (length > (uint32_t)(-1))) /* Check if length greater than UINT_MAX */
+  {
+    ret_value = ecma_make_throw_obj_completion_value (ecma_new_standard_error (ECMA_ERROR_RANGE));
```

Comment:
@dbatyai, the error should be raised in `ecma_op_array_object_define_own_property`. Is there any case, when it isn't raised?